src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs

```diff
@@ -40,22 +46,19 @@ public static OSPlatform Create(string osPlatform)
 
         public bool Equals(OSPlatform other)
         {
-            return Equals(other._osPlatform);
-        }
-
-        internal bool Equals(string? other)
-        {
-            return string.Equals(_osPlatform, other, StringComparison.Ordinal);
+            return string.Equals(_osPlatform, other._osPlatform, StringComparison.OrdinalIgnoreCase);
```

Comment:
What is motivating this change? This will make this API quite a bit more expensive.